core/unit_test/TestHalfOperators.hpp

```diff
@@ -177,83 +420,267 @@ struct Functor_TestHalfOperators {
     expected_lhs(CADD_H_H) = d_lhs;
     expected_lhs(CADD_H_H) += d_rhs;
 
-    // tmp_lhs = h_lhs;
-    // tmp_lhs += static_cast<float>(d_rhs);
-    // actual_lhs(CADD_H_S)   = cast_from_half<double>(tmp_lhs);
-    // expected_lhs(CADD_H_S) = d_lhs;
-    // expected_lhs(CADD_H_S) += d_rhs;
+    tmp_lhs = h_lhs;
+    tmp_lhs += static_cast<float>(d_rhs);
```

Comment:
Note: These compound expressions are expected to downcast the result due to the lhs of type `half_t`.

TODO: Add overloads or templates for `float` and `double` to ensure operands are promoted.